bindings/java/src/org/sleuthkit/datamodel/OsAccountManager.java

```diff
@@ -826,29 +809,58 @@ private String makeOsAccountUpdateQuery(String tableName, OsAccount sourceAccoun
 
 	/**
 	 * Copy all fields from sourceAccount that are not set in
-	 * matchingDestAccount. Does not update the database and does not update
-	 * associated Person.
+	 * destAccount. 
+	 * 
+	 * Updates the dest account in the database.
 	 *
 	 * @param sourceAccount The source account.
 	 * @param destAccount   The destination account.
+	 * @param trans	Transaction to use for database operations.
+	 * 
+	 * @return OsAccount Updated account.
 	 */
-	private void mergeOsAccountObjects(OsAccount sourceAccount, OsAccount destAccount) throws TskCoreException {
+	private OsAccount  mergeOsAccountObjectsAndUpdateDestAccount(OsAccount sourceAccount, OsAccount destAccount, CaseDbTransaction trans) throws TskCoreException {
+		
+		OsAccount mergedDestAccount = destAccount;
+		
+		String destLoginName = null;
+		String destAddr = null;
+		
+		
 		// Copy any fields that aren't set in the destination to the value from the source account.
 		if (!destAccount.getLoginName().isPresent() && sourceAccount.getLoginName().isPresent()) {
-			destAccount.setLoginName(sourceAccount.getLoginName().get());
+			destLoginName = sourceAccount.getLoginName().get();
 		}
 
 		if (!destAccount.getAddr().isPresent() && sourceAccount.getAddr().isPresent()) {
-			destAccount.setAddr(sourceAccount.getAddr().get());
+			destAddr = sourceAccount.getAddr().get();
 		}
 
+		// update the dest account core 
+		AccountUpdateStatus updateStatus = this.updateOsAccountCore(destAccount, destAddr, destLoginName, trans);
+		
+		if (updateStatus.getUpdateStatusCode() == AccountUpdateStatusEnum.UPDATED && updateStatus.getUpdatedAccount().isPresent()) {
+			mergedDestAccount = updateStatus.getUpdatedAccount().get();
+		}
+		
+		String destFullName = null;
+		Long destCreationTime = null;
 		if (!destAccount.getFullName().isPresent() && sourceAccount.getFullName().isPresent()) {
```

Comment:
It appears that updateOsAccountProperties will not change the field in destAccount if it exists, so the first half of this check may not longer be necessary. This is likely true of several other conditionals in this method as well.

Reply:
We've changed updateOsAccountProperties () to set any non-null value it is given.